pkg/feature/feature.go

```diff
@@ -46,17 +50,29 @@ func newFeature(name string) *Feature {
 // Action is a func type which can be used for different purposes while having access to Feature struct.
 type Action func(feature *Feature) error
 
-func (f *Feature) Apply() error {
+//nolint:nonamedreturns // Reason: we use the named return to handle errors in a unified fashion through deferred function.
+func (f *Feature) Apply() (err error) {
 	if !f.Enabled {
 		return nil
 	}
 
-	if err := f.createResourceTracker(); err != nil {
-		return err
+	if trackerErr := f.createFeatureTracker(); err != nil {
+		return trackerErr
 	}
 
 	// Verify all precondition and collect errors
 	var multiErr *multierror.Error
+	phase := featurev1.FeatureCreated
+	f.updateFeatureTrackerStatus(conditionsv1.ConditionDegraded, "False", phase, fmt.Sprintf("Applying feature %s", f.Name))
```

Comment:
why here is set to conditionsv1.ConditionDegraded , 
should not be ConditionProgressing ?

Reply:
Good spot. Lost between the lines.

```suggestion
	f.updateFeatureTrackerStatus(conditionsv1.ConditionProgressing, "False", phase, fmt.Sprintf("Applying feature %s", f.Name))
```